In [22]:
import pandas as pd

In [23]:
pd.read_csv('../../data/raw/course_offerings.csv')

,course_id,course_name,instructor
0,101,Intro to Python,Chris Bruehl
1,102,Intro to SQL,John Pauler
2,201,Exploratory Data Analysis,Alice Zhao
3,301,Algorithms,Chris Bruehl
4,331,Natural Language Processing,Alice Zhao


In [24]:
pd.read_excel('../../data/raw/Course Offerings.xlsx', sheet_name=1)

,Course ID,Course Name,Instructor
0,101,Intro to Python,Chris Bruehl
1,102,Intro to SQL,John Pauler
2,201,Exploratory Data Analysis,Alice Zhao
3,301,Algorithms,Chris Bruehl
4,331,Natural Language Processing,Alice Zhao


In [25]:
import sqlite3

In [26]:
conn = sqlite3.connect('../../data/raw/maven.db')

In [27]:
pd.read_sql('SELECT * FROM courses', conn)

,course_id,course,instructor
0,101,Intro to Python,Chris Bruehl
1,102,Intro to SQL,John Pauler
2,201,Exploratory Data Analysis,Alice Zhao
3,301,Algorithms,Chris Bruehl
4,331,Natural Language Processing,Alice Zhao


##Exploring a Dataframe

In [28]:
df = pd.read_csv('../../data/raw/course_offerings.csv')

In [29]:
df

,course_id,course_name,instructor
0,101,Intro to Python,Chris Bruehl
1,102,Intro to SQL,John Pauler
2,201,Exploratory Data Analysis,Alice Zhao
3,301,Algorithms,Chris Bruehl
4,331,Natural Language Processing,Alice Zhao


In [30]:
df.head(2)

,course_id,course_name,instructor
0,101,Intro to Python,Chris Bruehl
1,102,Intro to SQL,John Pauler


In [31]:
df.tail(2)

,course_id,course_name,instructor
3,301,Algorithms,Chris Bruehl
4,331,Natural Language Processing,Alice Zhao


In [32]:
df.shape

(5, 3)

In [33]:
df.count()

course_id      5
course_name    5
instructor     5
dtype: int64

In [34]:
df.describe()

,course_id
count,5.000000
mean,207.200000
std,107.829495
min,101.000000
25%,102.000000
50%,201.000000
75%,301.000000
max,331.000000


In [35]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   course_id    5 non-null      int64
 1   course_name  5 non-null      str  
 2   instructor   5 non-null      str  
dtypes: int64(1), str(2)
memory usage: 252.0 bytes


#ASSIGNMENT: READ A FILE INTO PYTHON

In [37]:
df_happiness_survey = pd.read_csv('../../data/raw/happiness_survey_data.csv')

In [38]:
df_happiness_survey

,country_name,year,happiness_score,social_support,freedom_to_make_life_choices,healthy_life_expectancy_at_birth
0,Afghanistan,2008,3.723590,0.450662,0.718114,50.500000
1,Afghanistan,2009,4.401778,0.552308,0.678896,50.799999
2,Afghanistan,2010,4.758381,0.539075,0.600127,51.099998
3,Afghanistan,2011,3.831719,0.521104,0.495901,51.400002
4,Afghanistan,2012,3.782938,0.520637,0.530935,51.700001
...,...,...,...,...,...,...
2084,Zimbabwe,2017,3.638300,0.754147,0.752826,52.150002
2085,Zimbabwe,2018,3.616480,0.775388,0.762675,52.625000
2086,Zimbabwe,2019,2.693523,0.759162,0.631908,53.099998
2087,Zimbabwe,2020,3.159802,0.717243,0.643303,53.575001


In [39]:
df_happiness_survey.shape

(2089, 6)

In [40]:
df_happiness_survey.describe()

,year,happiness_score,social_support,freedom_to_make_life_choices,healthy_life_expectancy_at_birth
count,2089.000000,2089.000000,2076.000000,2057.000000,2031.000000
mean,2013.727621,5.473747,0.811542,0.745462,63.180326
std,4.455614,1.115567,0.118935,0.140751,6.948546
min,2005.000000,2.178809,0.290184,0.257534,6.720000
25%,2010.000000,4.651972,0.747664,0.651689,58.965000
50%,2014.000000,5.405246,0.834770,0.767357,64.980003
75%,2017.000000,6.294282,0.904682,0.857677,68.362499
max,2021.000000,8.018934,0.987343,0.985178,74.349998


In [41]:
df_happiness_survey.info()

<class 'pandas.DataFrame'>
RangeIndex: 2089 entries, 0 to 2088
Data columns (total 6 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   country_name                      2089 non-null   str    
 1   year                              2089 non-null   int64  
 2   happiness_score                   2089 non-null   float64
 3   social_support                    2076 non-null   float64
 4   freedom_to_make_life_choices      2057 non-null   float64
 5   healthy_life_expectancy_at_birth  2031 non-null   float64
dtypes: float64(4), int64(1), str(1)
memory usage: 98.1 KB


#RESULTADO DEL ASSIGNMENT

* Number of Rows = 2089
* Range of the happiness Score = 2.178809 a 8.018934	